core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/NodeRequirements.java

```diff
@@ -33,14 +30,14 @@ public class NodeRequirements
     private static final int INSTANCE_SIZE = instanceSize(NodeRequirements.class);
 
     private final Optional<CatalogHandle> catalogHandle;
-    private final Set<HostAddress> addresses;
+    private final Optional<HostAddress> address;
```

Comment:
Was this supposed to map to `io.trino.spi.connector.ConnectorSplit#getAddresses` ?
I don't follow why this can only be 0 or 1 and not a list

Reply:
We pick a candidate from potentially longer list when we assign split to taskDescriptor in FTE. 
In practice I think it is always 1 or 0 anyway.